### PR TITLE
[CUDA] Fix random generator routines for f32 and f64 to make sure the returned value is in [0, 1)

### PR DESCRIPTION
**Pull request by neozhaoliang**
Currently the return value by `ti.random(f32)` (or  `ti.random(f64)` ) has a chance to be equal to 1.0, due to float precision problem. This PR fixes this problem.

### CODE REVIEW

**Review by qiao-bo — APPROVED** [review]
This works great! Thanks

**Review by qiao-bo — COMMENTED** [review]
just to undo the approve~

File: python/taichi/tools/image.py [hunk 1]
```diff
@@ -113,26 +91,36 @@ def imread(filename, channels=0):
     return img.swapaxes(0, 1)[:, ::-1, :]
 
 
-def imshow(img, window_name='imshow'):
-    """Show image in a Taichi GUI.
+def imshow(img, title='imshow'):
+    """Display a taichi.field or a numpy.ndarray in a Taichi GUI window or an interative Ipython notebook.
```
Comment (qiao-bo) [inline python/taichi/tools/image.py:95]:
```suggestion
    """Display a taichi.field or a numpy.ndarray in a Taichi GUI window or an interactive Ipython notebook.
```

**Review by qiao-bo — APPROVED** [review]
Looks great, just one typo. Thanks!

**Comment by neozhaoliang** [comment]
The typo is fixed.

**Pull request merged**